lib/game.rb

```diff
@@ -0,0 +1,167 @@
+require 'pry'
+##
+# => Game handles the functionality of the Tic-Tac-Toe game
+##
+class Game
+  # => creates getters and setters for the instance variables
+  attr_accessor :board, :player_1, :player_2, :num_players
+
+  ##
+  # => Instantiates a game, defaults are two players and an empty board
+  ##
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1;
+    @player_2 = player_2;
+    @board = board;
+  end
+
+  # => All of the possible winning combinations
+  WIN_COMBINATIONS = [
+    [0, 1, 2], # Top row
+    [3, 4, 5], # Middle row
+    [6, 7, 8], # Bottom row
+    [0, 3, 6], # First column
+    [1, 4, 7], # Second column
+    [2, 5, 8], # Third Column
+    [0, 4, 8], # Left to Right diagonal
+    [2, 4, 6]  # Right to Left diagonal
+  ];
+
+  ##
+  # => Asks how many human players and sets the players up accordingly
+  ##
+  def define_players
+    puts "How many players want to play? (0, 1, or 2)"
+    @num_players = gets.strip.to_i
+
+    get_player_input = "Would you like to be 'X's or 'O's?";
+    #binding.pry
+    case @num_players
+      when 0
+          @player_1 = Players::Computer.new("X")
+          @player_2 = Players::Computer.new("O")
+      when 1
+        puts get_player_input;
+        input = gets.strip;
+
+        input == "O" || input == "o" ? @player_1.token = "O" && @player_2 = Players::Computer.new("X") : @player_2 = Players::Computer.new("O")
+      when 2
+        puts get_player_input;
+        input = gets.strip;
+        if input == "O" || input == "o"
+          @player_1.token = "O" && @player_2.token = "X"
+        end
+      else
+        puts "Invalid input, please select a valid option.";
+        define_players;
+      end
+  end #end #define_players
+
+  ##
+  # => This figures out who's turn it is
+  ##
+  def current_player
+    @board.turn_count % 2 == 0 ? @player_1 : @player_2;
+  end
+
+  ##
+  # => This checks if somebody has won the game or not
+  ##
+  def won?
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.detect do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```

Reply:
What is the @board.taken?(combo[0] +1) doing?

I'm not sure what the reason for the +1 is...